### ISSUE TITLE
REQUEST: cursor stays hidden while scrolling

### ISSUE THREAD

**Issue by k-wax** [Airblader/unclutter-xfixes#13]
I mainly use two-finger-scroll on my laptop so the cursor doesn't move but scrolling still makes the cursor visible.

Original unclutter keeps the cursor hidden while scrolling but that seems to have an effect on scroll speed (slower) on applications such as spotify, soundnode and franz.

If there's a way to have a hidden cursor and normal, fast scroll speed i'd very much appreciate it if it was implemented. Thanks!

**Comment by Airblader** [comment]
We'd have to know which button was pressed, which should be possible. I'll look into this, but will be on vacation for the next two weeks.

**Comment by Airblader** [comment]
@k-wax Can you try the following patch and see if it works?

````
diff --git a/src/event.c b/src/event.c
index 1b53d08..460fec9 100644
--- a/src/event.c
+++ b/src/event.c
@@ -56,9 +56,23 @@ static void x_check_cb(EV_P_ ev_check *w, int revents) {
             continue;
         }
 
-        /* We don't actually need the data, so get rid of it. */
+        bool is_scrollwheel = false;
+        switch (cookie->evtype) {
+            case XI_RawButtonPress: ;
+                const XIRawEvent *data = (const XIRawEvent *) cookie->data;
+                is_scrollwheel = data->detail == 4 || data->detail == 5;
+                break;
+            default:
+                break;
+        }
+
         XFreeEventData(display, cookie);
 
+        // TODO XXX Make this configurable
+        if (is_scrollwheel) {
+            continue;
+        }
+
         if (config.jitter > 0 && cookie->evtype == XI_RawMotion) {
             Window root, child;
             int root_x, root_y, win_x, win_y;

```

It does work for me in principle, but I can't really manage to use two-finger scroll on my machine without triggering at least one button press for a normal "tap". I'm not sure if that's just me, though.

**Comment by k-wax** [comment]
It does exactly what i need, perfect! Thank you so much. Life is better now.

**Comment by Airblader** [comment]
Great, thanks for the feedback. I'll wrap this into a configurable option and do a release in a couple weeks when I'm back.

**Comment by Airblader** [comment]
@k-wax I've added this as `--ignore-scrolling` now and published a new minor release 1.2.
